cdap-security/src/main/java/io/cdap/cdap/security/tools/AccessTokenGeneratorService.java

```diff
@@ -0,0 +1,216 @@
+      .setHttpHandlers(handler);
+    if (cConf.getBoolean(Constants.Security.SSL.INTERNAL_ENABLED)) {
+      SConfiguration sConf = injector.getInstance(SConfiguration.class);
+      new HttpsEnabler().configureKeyStore(cConf, sConf).enable(builder);
+    }
+    httpService = builder.build();
+  }
+
+  @VisibleForTesting
+  InetSocketAddress getBindAddress() {
+    return httpService.getBindAddress();
+  }
+
+  @VisibleForTesting
+  TokenManager getTokenManager() {
+    return handler.tokenManager;
+  }
+
+  @VisibleForTesting
+  Codec<AccessToken> getTokenCodec() {
+    return handler.tokenCodec;
+  }
+
+  @Override
+  protected void startUp() throws Exception {
+    httpService.start();
+  }
+
+  @Override
+  protected void shutDown() throws Exception {
+    httpService.stop();
+  }
+
+  @Path(Constants.Gateway.INTERNAL_API_VERSION_3 + "/accesstoken")
+  public static class TokenGeneratorHttpHandler extends AbstractHttpHandler {
+    private final CConfiguration cConf;
+    final TokenManager tokenManager;
+    final Codec<AccessToken> tokenCodec;
+
+    @Inject
+    public TokenGeneratorHttpHandler(CConfiguration cConf, TokenManager tokenManager,
+                                     Codec<AccessToken> tokenCodec) {
+      this.cConf = cConf;
+      this.tokenManager = tokenManager;
+      this.tokenCodec = tokenCodec;
+    }
+
+    @Override
+    public void init(HandlerContext context) {
+      tokenManager.startUp();
+    }
+
+    @Override
+    public void destroy(HandlerContext context) {
+      tokenManager.stopAndWait();
+    }
+
+    @GET
+    @Path("/")
+    public void getToken(FullHttpRequest request, HttpResponder responder,
+                         @QueryParam("username") String username,
+                         @QueryParam("identifierType") @DefaultValue("INTERNAL") String identifierType)
+      throws IOException, BadRequestException {
+      UserIdentity.IdentifierType type;
+      try {
+        type = UserIdentity.IdentifierType.valueOf(identifierType);
+      } catch (IllegalArgumentException e) {
+        throw new BadRequestException("Invalid identifierType. Valid values: INTERNAL, EXTERNAL");
+      }
+      long tokenExpiration = cConf.getLong(Constants.Security.TOKEN_EXPIRATION);
+      long issueTime = System.currentTimeMillis();
+      long expireTime = issueTime + tokenExpiration;
+      UserIdentity userIdentity = new UserIdentity(username, type,
+                                                   Collections.emptyList(),
+                                                   issueTime, expireTime);
+      AccessToken token = tokenManager.signIdentifier(userIdentity);
+      String encodedToken = Base64.getEncoder().encodeToString(tokenCodec.encode(token));
+      responder.sendString(HttpResponseStatus.OK, encodedToken);
+    }
+  }
+
+  public static void main(String[] args)throws Exception {
+    Option portOption = new Option("p", "port", true, "Server port.");
```

Comment:
Can we make the token expiration, KeyManager implementation, and key filepath (for FileBasedKeyManager) overwritable via either command line args or environment variables?

Reply:
Discussed offline. Token expiration is now configurable via command line. Key manager implementation and file path are not because they need to be in sync with the router - cconf is the source of truth for this